tiledb/sm/tile/tile.h

```diff
@@ -160,6 +182,11 @@ class TileBase {
 
   /** The tile data type. */
   Datatype type_;
+
+  /**
+   * Whether to block waiting for io data to be ready before accessing data()
+   */
+  const bool skip_waiting_on_io_task_;
```

Comment:
`TIleBase` is where this field is declared, yet `TileBase` itself does not have the I/O tasks.  Is there a reason that this is here instead of in `class Tile`?

Similarly... is there ever a situation where the I/O task is valid and `skip_waiting_on_io_task_` is true?  Would `std::optional<ThreadTask::SharedTask>` be a more expressive way of doing the same thing?

Reply:
Sounds like a way better way. Let me try to make it an optional.

Reply:
As for way this is in `TileBase` and not in `Tile`, the reason is that it's also needed in `WriterTile` that inherits from `TileBase`